fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(data => this.setState({ todos: data }))
+      .catch(err => console.error(err));
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(res => res.json())
+      .then(this.state.todos.push(newTodo.task));
```

Comment:
Few things here, the push method will get called right away and its return value will be passed into .then. .then should be given a callback function to be called once the promise has resolved. Also you never want to directly change state, by calling push directly on `this.state.todos` you are mutating state which you always want to avoid. If you want to update an array in state, you need to first copy the array, make any changes to the copy, then use the copy to setState. Here is a reference on making a copy of an array:

https://www.samanthaming.com/tidbits/35-es6-way-to-clone-an-array/